site/src/job_queue/mod.rs

```diff
@@ -345,15 +353,51 @@ mod tests {
         }
     }
 
-    async fn mark_as_completed(conn: &dyn database::pool::Connection, shas: &[&str]) {
-        let completed_at = Utc::now();
-        for sha in shas {
-            conn.update_benchmark_request_status(
-                sha,
-                BenchmarkRequestStatus::Completed { completed_at },
-            )
+    async fn complete_request(
+        db: &dyn database::pool::Connection,
+        request_tag: &str,
+        collector_name: &str,
+        benchmark_set: u32,
+        target: &Target,
+    ) {
+        /* Create job for the request */
+        db.enqueue_benchmark_job(
+            request_tag,
+            &target,
+            &CodegenBackend::Llvm,
+            &Profile::Opt,
+            benchmark_set,
+        )
+        .await
+        .unwrap();
+
+        let job = db
+            .dequeue_benchmark_job(collector_name, &target, &BenchmarkSet(benchmark_set))
             .await
+            .unwrap()
             .unwrap();
+
+        assert_eq!(job.request_tag(), request_tag);
+
+        /* Mark the job as complete */
+        db.mark_benchmark_job_as_completed(job.id(), &BenchmarkJobConclusion::Success)
+            .await
+            .unwrap();
+
+        db.mark_benchmark_request_as_completed(request_tag)
```

Comment:
Same here.

Btw there's quite a lot of duplication between the DB and website tests around the job queue and requests. I'll go through them and try to create some more high-level test API and unify the tests in one of the packages, otherwise it will become quite cumbersome to modify the tests after code changes.